### PR TITLE
Feature/test document one shard

### PR DESCRIPTION
**Pull request by mchacki**
This adds a test for the upgrade of oneShardDatabase to support DOCUMENT method after upgrading.
This is known to be broken for 3.6 -> 3.7.5 but should be good after.

### CODE REVIEW

**Review by fceller — APPROVED** [review]
LGTM

File: test_data/makedata.js [hunk 1]
```diff
@@ -146,6 +146,46 @@ function createIndexSafe(options) {
...
+               return db._properties().sharding === "single";
+             }
+            );
+  if (!created) {
+    // its already wrongly there - skip this one.
+    print(`skipping ${databaseName} - it failed to created, but it is no one-shard.`);
+    return;
+  }
+  progress('created OneShard DB');
+  for (let ccount = 0; ccount < options.collectionMultiplier; ++ccount) {
```
Comment (dothebart) [inline test_data/makedata.js:180]: please use bulk operations to save the documents.

**Review by dothebart — CHANGES_REQUESTED** [review]
pls use bulk operations.

File: test_data/makedata.js [hunk 2]
```diff
@@ -146,6 +146,46 @@ function createIndexSafe(options) {
...
+               db._useDatabase(dbname);
+               return true;
+             }, dbname => {
+               db._useDatabase(dbname);
+               return db._properties().sharding === "single";
+             }
+            );
+  if (!created) {
+    // its already wrongly there - skip this one.
+    print(`skipping ${databaseName} - it failed to created, but it is no one-shard.`);
```
Comment (mchacki) [inline test_data/makedata.js:176]:
```suggestion
    print(`skipping ${databaseName} - it failed to be created, but it is no one-shard.`);
```

**Review by dothebart — APPROVED** [review]

**Pull request merged**